liualgotrader/data/alpaca.py

```diff
@@ -392,6 +392,10 @@ def get_symbol_data(
                 tlog("retrying")
                 time.sleep(10)
                 return self.get_symbol_data(symbol, start, end, scale)
+            else:
+                raise ValueError(
+                    f"[EXCEPTION] {e} for {symbol} could not obtain data for {_start} to {_end} w {scale.name}"
+                )
```

Comment:
just adding my 5 cents here: You **could** also add [exception chaining](https://peps.python.org/pep-3134/) which was introduced in python3.0. So `raise ValueError(...) from e`. But since no other place seems to follow that I have left that out.